### PR TITLE
Scala operators.json details added.

### LINKED ISSUE #310: Add Scala Logic/Math Operators
# Description

Logic/Math Operators for Scala aren't presently on Code Thesaurus. This issue is to add that concept to the system.

# Requirements

* Copy `web/thesauruses/_meta/operators.json` to the right Scala directory in `web/thesauruses/`
* Edit the operators.json file to accurately reflect the latest major version of Scala
* Delete the `"name":` fields before your PR
* Add in `"comment":` if you wish to explain something in the code
* Add in `"not-implemented": "true"` if the concept doesn't exist in the language. You can then leave `"code":` blank.

# Claiming This Issue

Please leave a comment below so I can assign you to it. This is to prevent multiple people from working on the same issue. 

Check out [CONTRIBUTING.md](https://github.com/codethesaurus/codethesaur.us/blob/main/CONTRIBUTING.md) for more information on how to contribute. 

Check out [https://docs.codethesaur.us/thesaurus/](https://docs.codethesaur.us/thesaurus/) for information on how Code Thesaurus works and how to add to the thesauruses.

# Need Help?

Leave a comment on here with your question, or feel free to ask [@geekygirlsarah](https://twitter.com/geekygirlsarah) on Twitter. You can also email coreteam@codethesaur.us too.

# Thanks!

We really value and appreciate the additions people make to Code Thesaurus, no matter how large or small. So thank you for taking the time to work on this!

### PR DESCRIPTION
**Pull request by chaitanyks**
## What GitHub issue does this PR apply to?

Resolves #310 

## Checklist

- [x] I claimed any associated issue(s) and they are not someone else's
- [ ] I have looked at documentation to ensure I made any revisions correctly
- [x] I tested my changes locally to ensure they work
- [ ] (If editing Django) I have added or edited any appropriate unit tests for my changes

### CODE REVIEW

**Comment by chaitanyks** [comment]
Scala_image

Changes are working fine.

File: web/thesauruses/scala/operators.json [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+      "right_zero_fill"
+    ],
+    "Conditional Operators": [
+      "ternary",
+      "null_forgiving"
+    ]
+  },
+  "operators": {
+    "addition": {
+      "name": "Addition operator",
```
Comment (geekygirlsarah) [inline web/thesauruses/scala/operators.json]: I saw you didn't look over the documentation, but in the language files, you should take out the `name` lines as those are pulled in from the meta files and aren't needed here.

File: web/thesauruses/scala/operators.json [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+      "code": "/"
+    },
+    "division_assignment": {
+      "name": "Division and assignment operator",
+      "code": "/="
+    },
+    "integer_division": {
+      "not-implemented": "true",
+      "name": "Integer division operator",
+      "code": ""
```
Comment (geekygirlsarah) [inline web/thesauruses/scala/operators.json]: For the `not-implemented` concepts, can you remove the `code` lines? It just cleans up the file when no code is needed.

File: web/thesauruses/_meta/operators.json [hunk 3]
```diff
@@ -49,10 +49,14 @@
       "xor_assignment",
       "xnor",
       "xnor_assignment",
+      "binary_and",
```
Comment (geekygirlsarah) [inline web/thesauruses/_meta/operators.json]: This issue is _just_ for editing Scala's language operators. The meta ones shouldn't be edited in the same PR. But also the binary AND and OR and etc. are the bitwise ones above (`and` and `or`) so these are duplicates. The one's compliment and zero fill might be good additions but should be reported in a separate issue.

Can you remove these lines you added to take this file back to its original state, and then remove the added concepts in the Scala operators file too?

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]
I think with a few changes, I'd be happy to merge this in. I'll make an issue for the new operators to edit the meta one. You're welcome to take that if you want or I'll add it in possibly tonight or tomorrow.

File: web/thesauruses/_meta/operators.json (hunk 3 above)
Comment (chaitanyks) [reply web/thesauruses/_meta/operators.json]: Yaa sure.

File: web/thesauruses/scala/operators.json (hunk 1 above)
Comment (chaitanyks) [reply web/thesauruses/scala/operators.json]: Fixed.

File: web/thesauruses/scala/operators.json (hunk 2 above)
Comment (chaitanyks) [reply web/thesauruses/scala/operators.json]: Fixed.

**Comment by geekygirlsarah** [comment]
With the changes, I think this looks good! I'll merge it in.

It counts towards Hacktoberfest, so be sure to sign up for that if you haven't to get a shirt/swag!

**Pull request merged**